fabric-model-loading-api-v1/src/testmodClient/java/net/fabricmc/fabric/test/model/loading/ModelTestModClient.java

```diff
@@ -0,0 +1,122 @@
+
+		@Override
+		public void emitBlockQuads(BlockRenderView blockView, BlockState state, BlockPos pos, Supplier<Random> randomSupplier, RenderContext context) {
+			context.pushTransform(q -> q.cullFace() != Direction.DOWN);
+			super.emitBlockQuads(blockView, state, pos, randomSupplier, context);
+			context.popTransform();
+		}
+	}
+
+	@Override
+	public void onInitializeClient() {
+		ModelLoadingPlugin.register(pluginContext -> {
+			pluginContext.addModels(MODEL_ID);
+			// remove bottom face of gold blocks
+			pluginContext.modifyModelAfterBake().register(ModelModifier.WRAP_PHASE, (model, context) -> {
+				if (context.id().getPath().equals("block/gold_block")) {
+					return new DownQuadRemovingModel(model);
+				} else {
+					return model;
+				}
+			});
+			// make fences with west: true and everything else false appear to be a missing model visually
+			ModelIdentifier fenceId = BlockModels.getModelId(Blocks.OAK_FENCE.getDefaultState().with(HorizontalConnectingBlock.WEST, true));
+			pluginContext.modifyModelOnLoad().register(ModelModifier.OVERRIDE_PHASE, (model, context) -> {
+				if (fenceId.equals(context.id())) {
+					return context.getOrLoadModel(ModelLoader.MISSING_ID);
+				}
+
+				return model;
+			});
+			// make brown glazed terracotta appear to be a missing model visually, but without affecting the item, by using pre-bake
+			// using load here would make the item also appear missing
+			pluginContext.modifyModelBeforeBake().register(ModelModifier.OVERRIDE_PHASE, (model, context) -> {
+				if (context.id().getPath().equals("block/brown_glazed_terracotta")) {
+					return context.loader().getOrLoadModel(ModelLoader.MISSING_ID);
+				}
+
+				return model;
+			});
+
+			// Make wheat stages 1->6 use the same model as stage 0. This can be done with resource packs, this is just a test.
+			pluginContext.registerBlockStateResolver(Blocks.WHEAT, context -> {
+				BlockState state = context.block().getDefaultState();
+
+				// All the block state models are top-level...
+				// Use a delegating unbaked model to make sure the identical models only get baked a single time.
+				Identifier wheatStage0Id = new Identifier("block/wheat_stage0");
+
+				UnbakedModel stage0Model = new DelegatingUnbakedModel(wheatStage0Id);
+
+				for (int age = 0; age <= 6; age++) {
+					context.setModel(state.with(CropBlock.AGE, age), stage0Model);
+				}
+
+				context.setModel(state.with(CropBlock.AGE, 7), context.getOrLoadModel(new Identifier("block/wheat_stage7")));
+			});
+		});
+
+		ResourceManagerHelper.get(ResourceType.CLIENT_RESOURCES).registerReloadListener(SpecificModelReloadListener.INSTANCE);
+
+		LivingEntityFeatureRendererRegistrationCallback.EVENT.register((entityType, entityRenderer, registrationHelper, context) -> {
+			if (entityRenderer instanceof PlayerEntityRenderer) {
+				registrationHelper.register(new BakedModelFeatureRenderer<>((PlayerEntityRenderer) entityRenderer, SpecificModelReloadListener.INSTANCE::getSpecificModel));
```

Comment:
```suggestion
			if (entityRenderer instanceof PlayerEntityRenderer playerRenderer) {
				registrationHelper.register(new BakedModelFeatureRenderer<>(playerRenderer, SpecificModelReloadListener.INSTANCE::getSpecificModel));
```
We are not on Java 8 anymore